lib/collection.js

```diff
@@ -0,0 +1,291 @@
+var Events = require('./events'),
+    Collection,
+    ARRAY_PROXY_METHODS = ['forEach', 'some', 'every', 'filter', 'map', 'reduce'];
+
+function clone(obj) {
+    return Object.keys(obj).reduce(function (result, key) {
+        result[key] = obj[key];
+        return result;
+    }, {});
+}
+
+/**
+ * @class Collection
+ * @extends Events
+ */
+Collection = Events.inherit(/** @lends Collection.prototype*/{
+
+    /**
+     * @type {Function}
+     */
+    modelType: null,
+
+    /**
+     * @type {Number}
+     */
+    length: 0,
+
+    /**
+     * @param {Array.<(Model|Object)>} [data]
+     */
+    __constructor: function (data) {
+        this.__base.apply(this, arguments);
+
+        /**
+         * @type {Array.<Model>}
+         */
+        this._models = [];
+
+        data = data || [];
+        this.set(data);
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Boolean}
+     */
+    isChanged: function (branch) {
+        return this._models.some(function (model) {
+            return model.isChanged(branch);
+        });
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Boolean}
+     */
+    commit: function (branch) {
+        return this._models.reduce(function (changed,  model) {
```

Comment:
it will be emited here https://github.com/bem-node/promised-models/pull/11/files#diff-a3644fc6e558da2bde71112d24ec75baR280